### PR TITLE
Migrate docs to `.com`

### PR DESCRIPTION
**Pull request by nankolena**
~~**!!! DO NOT MERGE until Github secrets are added**~~ (done)

Part of KOL-3716

### Linked issue(s)
https://linear.app/kolena/issue/KOL-3716/migrate-to-docskolenacom

### What change does this PR introduce and why?
- Update doc references to `docs.kolena.com`.
- Update doc release to new buckets hosting `.com` resources
- Fix some small typos along the way.

### Please check if the PR fulfills these requirements

- [x] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [x] Relevant docs have been added / updated

### CODE REVIEW

**Comment by yifanwu-kolena** [comment]
still seeing some instances of `docs.kolena.io`, is this intentional?

```sh
$ git grep docs.kolena.io
                                                                                                                                                                                                                    
.github/workflows/docs.yml:          aws s3 sync ./site "s3://trunk-docs.kolena.io" --delete
.github/workflows/docs.yml:          aws s3 sync ./site "s3://docs.kolena.io" --delete
.github/workflows/release.yml:          aws s3 sync ./site "s3://docs.kolena.io" --delete
CONTRIBUTING.md:The documentation for `kolena`, hosted at [docs.kolena.io](https://docs.kolena.io/), is built out of this repo using
README.md:  <a href="https://docs.kolena.io">
README.md:    <img src="https://docs.kolena.io/assets/images/kolena-banner.png" width="640" alt="Kolena" />
README.md:platform. [Install](https://docs.kolena.io/installing-kolena) with your favorite Python package manager and
README.md:[get started in minutes](https://docs.kolena.io/dataset/quickstart):
README.md:| [Developer Guide](https://docs.kolena.io) | [Quickstart](https://docs.kolena.io/dataset/quickstart) | [API Reference](https://docs.kolena.io/reference) | [Examples](./examples) |
docs/CNAME:docs.kolena.io
docs/overrides/main.html:      service: 'docs.kolena.io',
docs/overrides/main.html:      env: window.location.hostname === 'docs.kolena.io' ? 'production' : 'development',
mkdocs.yml:site_url: https://docs.kolena.io
pyproject.toml:documentation = "https://docs.kolena.io"
```

**Comment by nankolena** [comment]
> still seeing some instances of `docs.kolena.io`, is this intentional?
> 
> ```shell
> $ git grep docs.kolena.io
>                                                                                                                                                                                                                     
> .github/workflows/docs.yml:          aws s3 sync ./site "s3://trunk-docs.kolena.io" --delete
> .github/workflows/docs.yml:          aws s3 sync ./site "s3://docs.kolena.io" --delete
> .github/workflows/release.yml:          aws s3 sync ./site "s3://docs.kolena.io" --delete
> CONTRIBUTING.md:The documentation for `kolena`, hosted at [docs.kolena.io](https://docs.kolena.io/), is built out of this repo using
> README.md:  <a href="https://docs.kolena.io">
> README.md:    <img src="https://docs.kolena.io/assets/images/kolena-banner.png" width="640" alt="Kolena" />
> README.md:platform. [Install](https://docs.kolena.io/installing-kolena) with your favorite Python package manager and
> README.md:[get started in minutes](https://docs.kolena.io/dataset/quickstart):
> README.md:| [Developer Guide](https://docs.kolena.io) | [Quickstart](https://docs.kolena.io/dataset/quickstart) | [API Reference](https://docs.kolena.io/reference) | [Examples](./examples) |
> docs/CNAME:docs.kolena.io
> docs/overrides/main.html:      service: 'docs.kolena.io',
> docs/overrides/main.html:      env: window.location.hostname === 'docs.kolena.io' ? 'production' : 'development',
> mkdocs.yml:site_url: https://docs.kolena.io
> pyproject.toml:documentation = "https://docs.kolena.io"
> ```

Good call, made [a commit](https://github.com/kolenaIO/kolena/pull/510/commits/cbc8b868e61c749021acd2b36cc258459a52eef1) just now to clean up the rest. Now we are left with:
```
.github/workflows/docs.yml:          aws s3 sync ./site "s3://trunk-docs.kolena.io" --delete
.github/workflows/docs.yml:          aws s3 sync ./site "s3://docs.kolena.io" --delete
.github/workflows/release.yml:          aws s3 sync ./site "s3://docs.kolena.io" --delete
docs/CNAME:docs.kolena.io
```
The first 3 are left in place as we want to keep the `.io` buckets the same way for the near term, until we are comfortable redirecting all `.io` docs to `.com`. The CNAME file is not needed as it's handled by cloudfront.

Could you help reviewing [these two lines of change ](https://github.com/kolenaIO/kolena/pull/510/commits/cbc8b868e61c749021acd2b36cc258459a52eef1#diff-9c2d836b64a8163ccdd77c4928201aae0d01b69b97990008539f6ccadf885b95R20-R21)? The intention is to monitor in our existing DD RUM application:
- Service name will be `docs.kolena.io` or `docs.kolena.com`
- Env will be `development` or `production`.

File: docs/overrides/main.html [hunk 1]
```diff
@@ -17,8 +17,8 @@
       clientToken: '{{ config.extra.dd_rum.client_token }}',
       applicationId: '{{ config.extra.dd_rum.application_id }}',
       site: 'datadoghq.com',
-      service: 'docs.kolena.io',
-      env: window.location.hostname === 'docs.kolena.io' ? 'production' : 'development',
+      service: window.location.hostname.replace("trunk-docs", "docs"),
+      env: window.location.hostname.startsWith("docs") ? 'production' : 'development',
```
Comment (yifanwu-kolena) [inline docs/overrides/main.html:21]: i think this change LGTM

**Review by yifanwu-kolena — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -4,7 +4,7 @@ version = "0.999.0"  # version is automatically set to latest git tag during rel
 description = "Client for Kolena's machine learning testing platform."
 authors = ["Kolena Engineering <eng@kolena.com>"]
 homepage = "https://kolena.io"
```
Comment (gordonhart) [inline pyproject.toml]: Update to `kolena.com`?

**Pull request merged**